File: 86 (Partition List)/python/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def partition(self, head: Optional[ListNode], x: int) -> Optional[ListNode]:

        firstPartition = ListNode() # FP
        secondPartition = ListNode() # SP

        first = firstPartition 
        second = secondPartition

        while head:
            if head.val<x: # val < x
                first.next = head # first->next = head
                first = first.next # first = first.next
            else:
                second.next = head
                second = second.next
            
            head = head.next
        second.next = None
        first.next = secondPartition.next
        return firstPartition.next

File: 86 (Partition List)/java/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode partition(ListNode head, int x) {
        ListNode firstPartition = new ListNode();
        ListNode secondPartition = new ListNode();

        ListNode first = firstPartition;
        ListNode second = secondPartition;

        while (head!=null){
            if (head.val <x){
                first.next = head;
                first = first.next;
            }else{
                second.next = head;
                second = second.next;
            }
            head = head.next;
        }

        second.next = null;
        first.next = secondPartition.next;
        return firstPartition.next;
    }
}